index.js

```diff
@@ -812,6 +861,35 @@ SqlPort.prototype.callSP = function(name, params, flatten, fileName) {
                 }
             }
         });
+        if ($meta.saveAs) {
+            var fileDir = path.dirname($meta.saveAs);
+            return new Promise((resolve, reject) => {
+                fs.access(fileDir, (err) => {
+                    if (!err) {
+                        return resolve();
+                    }
+                    fs.mkdir(fileDir, (e) => {
+                        if (!e) {
+                            return resolve();
+                        }
+                        reject(e);
+                    });
+                });
+            }).then(function(resolve, reject) {
+                request.stream = true;
+                var ws = fs.createWriteStream(fileName);
+                request.execute(name);
+                transform(request).pipe(ws);
```

Comment:
I think you need to pipe before execute. At least their example is doing so: https://github.com/patriksimek/node-mssql#pipe-stream

Reply:
Please check it now